aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -28,6 +28,152 @@
 namespace at {
 namespace native {
 
+static
+void getDataOffsets(const TensorIteratorBase& iter, simd_uint3* data_offsets, uint32_t num_outputs) {
+  // Collect offsets for Input, Output and Index tensors
+  const uint32_t nOffsets = 3;
+  const uint32_t nDim = iter.ndim();
+
+  std::vector<uint32_t> strides[nDim];
```

Comment:
[Variable length arrays](https://en.wikipedia.org/wiki/Variable-length_array) are very weird feature and actually optional according to C++ standard.
Also `std::vector::push_back` is pretty expensive op, considering that nOffsets is known at compile time, better allocated it as `std::array<uint32_t, nOffsets>` and declare `nOffsets` as constexpr

And use `std::vector` with size passed to its constructor to avoid VLA
```suggestion
  constexpr uint32_t nOffsets = 3;
  const uint32_t nDim = iter.ndim();

  std::vector<std::array<uint32_t, nOffsets> strides(nDim);
```